packages/cadl-ranch-specs/http/clients/authentication/main.cadl

```diff
@@ -0,0 +1,26 @@
+import "@cadl-lang/rest";
+import "@azure-tools/cadl-azure-core";
+
+using Cadl.Http;
+using Azure.Core.Foundations;
+
+@serviceTitle("ClientAuthentication")
+@serviceVersion("1.0.0")
+@server("http://localhost:3000", "TestServer endpoint")
+@doc("Illustrates clients generated with OAuth2 and ApiKey authentication.")
+
+@useAuth(OAuth2Auth<[OAuth2AuthFlow]> | ApiKeyAuth<ApiKeyLocation.header, "x-ms-api-key">)
+namespace Cadl.TestServer.ClientAuthentication;
```

Comment:
```suggestion
namespace OAuth2Authentication;
```
this bad pattern of having the `Cadl.TestServer` namespace was started, I think we can just have the namespace be the scenario group we are testing.